magento/magento2
# [Backport] magento/magento2#9961: Unused product attributes display with value N/A or NO on storefront.

This is a backport for https://github.com/magento/magento2/pull/10619

### Description
Attributes with no value set should not display on storefront.

### Fixed Issues (if relevant)
1. magento/magento2#9961: Unused product attributes display with value N/A or NO on storefront.
2. magento/magento2#6634: Yes/No attribute value is not shown on a product details page

### Manual testing scenarios
1. Create a Custom Attribute,  set it's "Catalog Input Type for Store Owner" to "Text Area" ,  "Visible on Catalog Pages on Storefront" to "Yes".
2. Assign Custom Attribute to Default Attribute Set.
3. Create a Custom Product with Default Attribute Set, make it visible on Storefront, do not set any value to Custom Attribute.
4. Open Custom Product on Storefront - the Custom Attribute label and it's value should not be visible.
5. Edit the Custom Product, set any value to it's Custom Attribute.
6. Open Custom Product on Storefront - the Custom Attribute label and it's value should be visible.

### Contribution checklist
